spec/factories/conditions.rb

```diff
@@ -39,8 +39,8 @@
   factory :condition do
     op "eq"
     left_qing { build(:questioning) }
-    value { left_qing.has_options? ? nil : "1" }
-    option_node { left_qing.has_options? ? left_qing.option_set.c[0] : nil }
+    value { left_qing.present? && !left_qing.has_options? ? "1" : nil }
+    option_node { left_qing&.has_options? ? left_qing.option_set.c[0] : nil }
```

Comment:
In the two lines above there's `left_qing.has_options` and `left_qing&.has_options`, why the difference?

Reply:
`&.` is safe navigation, meaning if the object is `nil`, the method won't be called.

It's not needed on line 42 because there is an explicit check. `&.` is not used on that line because it's a negative condition. If you look at the truth table it doesn't work out right if you use `&.`.